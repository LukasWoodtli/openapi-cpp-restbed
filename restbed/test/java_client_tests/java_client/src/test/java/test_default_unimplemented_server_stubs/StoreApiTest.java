package test_default_unimplemented_server_stubs;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.openapi.example.api.StoreApi;
import org.openapi.example.model.Order;
import org.openapi.example.invoker.ApiClient;
import org.springframework.web.reactive.function.client.WebClientResponseException;

import static helper.ApiClientFactories.setUpStoreApi;
import static org.junit.jupiter.api.Assertions.*;
import static helper.TestingHelper.approveException;

class StoreApiTest {

    StoreApi apiInstance;

    @BeforeEach
    void setUp() {
        apiInstance = setUpStoreApi(1234);
    }

    @Test
    void deleteOrder() {
        var exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.deleteOrder("myOrderId").block();
                });

        approveException(exception);
    }

    @Test
    void getInventory() {
        var exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.getInventory().block();
                });

        approveException(exception);
    }

    @Test
    void getOrderById() {
        var exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.getOrderById(1L).block();
                });

        approveException(exception);
    }

    @Test
    void placeOrder() {
        var order = new Order();
        var exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.placeOrder(order).block();
                });

        approveException(exception);
    }
}