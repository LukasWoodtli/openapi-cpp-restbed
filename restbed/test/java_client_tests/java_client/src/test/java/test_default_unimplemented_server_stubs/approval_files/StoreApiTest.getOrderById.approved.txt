Status code: 404 NOT_FOUND
Reason: 
Response headers: []
